Client/Ecosia/Extensions/LegacyThemeManager+Ecosia.swift

```diff
@@ -3,19 +3,44 @@
 // file, You can obtain one at http://mozilla.org/MPL/2.0/
 
 import UIKit
+import Common
 
 extension LegacyThemeManager {
-    // Ecosia: theme changing on a central place
+    /// Updates the theme based on the changes in the trait collection.
+    ///
+    /// - Parameters:
+    ///   - from: The previous trait collection. This is optional as it may be nil when there's no previous state to compare to.
+    ///   - to: The new trait collection after the change. This is used to determine the current theme settings.
+    ///   - forceDark: A boolean indicating whether the dark theme should be forced on regardless of the system theme. The default value is `false`.
     func themeChanged(from: UITraitCollection?, to: UITraitCollection, forceDark: Bool = false) {
+        // Determine if the color appearance has changed between the previous and new trait collections.
         let colorHasChanged = to.hasDifferentColorAppearance(comparedTo: from)
 
-
-        // Do not change theme if it was dark before night mode already
+        // Return early and do not change the theme if the color hasn't changed, dark mode is forced,
+        // or the system theme is already in effect.
         guard colorHasChanged,
               !forceDark,
               systemThemeIsOn else { return }
 
+        // Update the current theme based on the new user interface style.
         let userInterfaceStyle = to.userInterfaceStyle
         current = userInterfaceStyle == .dark ? LegacyDarkTheme() : LegacyNormalTheme()
     }
 }
+
+extension LegacyThemeManager {
+    
+    /// Updates the current theme based on the system theme type provided by the `EcosiaThemeManager`.
+    ///
+    /// This method retrieves the `EcosiaThemeManager` from the shared `AppContainer` and updates
+    /// the theme accordingly. It checks whether the system theme type is dark and sets the
+    /// `current` theme to `LegacyDarkTheme` or `LegacyNormalTheme` based on that.
+    func updateBasedOnCurrentSystemThemeType() {
```

Comment:
Would it make sense to make this static since it only updates the shared instance? 🤔